gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/Oplog.java

```diff
@@ -3427,6 +3436,9 @@ else if (tag == null) {
             CompactionRecord cr = new CompactionRecord(keyBytes, crOffset);
             getRecoveryMap().put(oplogKeyId, cr);
             drs.getDiskRegionView().incRecoveredEntryCount();
+            if (EntryBits.isAnyInvalid(userBits) || EntryBits.isTombstone(userBits)) {
+              drs.getDiskRegionView().incInvalidOrTombstoneEntryCount();
+            }
```

Comment:
incRecoveredEntryCount does not bother about the type of value being recovered and for it even a tombstone recovered results in increment of recoveryCount. To match the recovered entry with indexes I had to keep this extra accounting. This counter goes hand in hand with the recoveredEntryCount.